src/repository.py

```diff
@@ -141,39 +148,76 @@ def is_dirty(self) -> bool:
         """
         return self._git_repo.is_dirty(untracked_files=True)
 
-    def get_file_content(self, path: str, branch: str | None = None) -> str:
-        """Get the content of a file from the default branch.
+    def tag_commit(self, tag_name: str, commit_sha: str) -> None:
```

Comment:
If we do this, we'll be tagging twice per commit, one by the upload charm actions and one by this action. I suggest we don't tag as part of the docs upload at all